packages/cursorless-engine/languages/perl.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  ancestorChainNodeMatcher,
+  cascadingMatcher,
+  createPatternMatchers,
+  matcher,
+  patternMatcher,
+} from "../util/nodeMatchers";
+import { NodeMatcherAlternative, SelectionWithEditor } from "../typings/Types";
+import { SimpleScopeTypeType } from "@cursorless/common";
+import { SyntaxNode } from "web-tree-sitter";
+import { getNodeRange, unwrapSelectionExtractor } from "../util/nodeSelectors";
+import { patternFinder } from "../util/nodeFinders";
+
+// Generated by the following command:
+// curl https://raw.githubusercontent.com/ganezdragon/tree-sitter-perl/ee1001210af5f32ba14d2ced834636548e1b6485/src/node-types.json \
+// | jq '[.[] | select(.type|match("_statement")) | .type ]'
+const STATEMENT_TYPES = [
+  "ellipsis_statement",
+  "for_simple_statement",
+  "for_statement_1",
+  "for_statement_2",
+  "foreach_simple_statement",
+  "foreach_statement",
+  "if_simple_statement",
+  "if_statement",
+  "loop_control_statement",
+  "loop_control_statement",
+  "named_block_statement",
+  "package_statement",
+  "pod_statement",
+  "require_statement",
+  "single_line_statement",
+  "unless_simple_statement",
+  "unless_statement",
+  "until_simple_statement",
+  "until_statement",
+  "use_constant_statement",
+  "use_no_feature_statement",
+  "use_no_if_statement",
+  "use_no_statement",
+  "use_no_subs_statement",
+  "use_parent_statement",
+  "when_simple_statement",
+  "while_simple_statement",
+  "while_statement",
+];
+
+const EXPRESSION_TYPES = [
```

Comment:
Do you have plans for these? They might not be necessary

Reply:
I'm not sure, I've basically been doing what the ruby code does (given C and Lisp provided important inspiration for Perl, and Perl and Lisp were among those that inspired Ruby).

Reply:
They're gone for the time being, so at least that makes this discussion irrelevant.